src/rgw/rgw_op.cc

```diff
@@ -6394,6 +6394,28 @@ int RGWCompleteMultipart::verify_permission(optional_yield y)
   rgw_iam_add_crypt_attrs(s->env, s->info.crypt_attribute_map);
 
   if (s->iam_policy || ! s->iam_user_policies.empty() || ! s->session_policies.empty()) {
+    if (s->bucket->get_info().obj_lock_enabled() && bypass_governance_mode) {
+      auto r = eval_identity_or_session_policies(this, s->iam_user_policies, s->env,
+                                  rgw::IAM::s3BypassGovernanceRetention, ARN(s->bucket->get_key(), s->object->get_name()));
```

Comment:
this does look plausibly similar to the logic in RGWDeleteObj::verify_permission, but I think I am looking for a matching call to verify_object_lock(...), probably in RGWCompleteMultipart::execute(...) that includes the updated value for bypass_perm, no?  I've looked for that in this PR, and don't see it.

Reply:
ok, I'm probably wrong here, we're not vulnerable to an overwrite in a versioned bucket